app/process/maxspeed/maxspeed.lua

```diff
@@ -0,0 +1,434 @@
+  if object.tags.highway == "steps" then
+    object.tags._skipNotes = object.tags._skipNotes .. ";Skipped `highway=steps`"
+    object.tags._skip = true
+  end
+
+  if object.tags.highway == "pedestrian" then
+    if object.tags.bicycle == "yes" then
+      object.tags._skip = false
+    else
+      object.tags._skipNotes = object.tags._skipNotes .. ";Skipped `highway=pedestrian + bicycle!=yes`"
+      object.tags._skip = true
+    end
+  end
+  --  verkehrsberuhigere bereich als string (nicht als Zahl)
+  if object.tags.highway == "living_street" and not object.tags.bicycle == "no" then
+    object.tags.category = "livingStreet"
+    object.tags.maxspeed = "Verkehrsberuhigter Bereich"
+    object.tags._skip = false
+  end
+
+  if object.tags.bicycle_road == "yes" or StartsWith(object.tags.traffic_sign, "DE:244") then
+    object.tags._skip = false
+  end
+
+  if (object.tags.bicycle == "designated" and object.tags.foot == "designated" and object.tags.segregated == "no") or
+      StartsWith(object.tags.traffic_sign, "DE:240")
+  then
+    object.tags.category = "footAndCycleway_shared"
+    object.tags._skip = false
+  end
+
+  if (object.tags.bicycle == "designated" and object.tags.foot == "designated" and object.tags.segregated == "yes") or
+      StartsWith(object.tags.traffic_sign, "DE:241")
+  then
+    object.tags.category = "footAndCycleway_segregated"
+    object.tags._skip = false
+  end
+
+  if object.tags.highway == "footway" or object.tags.highway == "path" then
+    if object.tags["mtb:scale"] then
+      object.tags._skipNotes = object.tags._skipNotes .. ";Skipped `highway=footway|path` but `mtb:scale`"
+      object.tags._skip = true
+    end
+
+    if object.tags.bicycle == "yes" or StartsWith(object.tags.traffic_sign, "DE:239,1022-10") then
+      object.tags.category = "footway_bicycleYes"
+      object.tags._skip = false
+    end
+  end
+end
+
+function Findmaxvalue(object)
+  if object.tags["maxspeed:forward"] then
+    object.tags.maxspeed = object.tags["maxspeed:forward"]
+    object.tags.source_maxspeed_forward = object.tags["source:maxspeed:forward"]
+    object.tags._todo = false
+    object.tags._skip = false
+  end
+  if object.tags["maxspeed:backward"] then
+    object.tags.maxspeed = object.tags["maxspeed:backward"]
+    object.tags.source_maxspeed_backward = object.tags["source:maxspeed:backward"]
+    object.tags._todo = false
+    object.tags._skip = false
+  end
+  local tbl = {}
+  -- Pick the highest value if multiple present.
+  if (object.tags["maxspeed:forward"]) then
+    table.insert(tbl, tonumber(object.tags["maxspeed:forward"]))
+  else
+    table.insert(tbl, -999)
+  end
+  if (object.tags["maxspeed:backward"]) then
+    table.insert(tbl, tonumber(object.tags["maxspeed:backward"]))
+  else
+    table.insert(tbl, -999)
+  end
+  if (object.tags["maxspeed"]) then
+    table.insert(tbl, tonumber(object.tags["maxspeed"]))
```

Comment:
probably we need to replace `tonumber()` with a version which remove possible units from the value. E.g. `20kmh | 20 kmh` -> `20`

Reply:
now handled by osm2pgsql because we pass the value directly into the db